backend/src/database/daos/betas/betas.dao.service.ts

```diff
@@ -2,16 +2,66 @@ import { Inject, Injectable } from '@nestjs/common';
 import { ModelClass } from 'objection';
 import { BetaModel } from '../../models/beta.model';
 import CreateBetaDto from '../../../betas/dtos/createBeta.dto';
+import { GetBetasQuery } from '../../../betas/dtos/getBetas.dto';
+
+type CreateBetaQuery = CreateBetaDto & {
+  creatorId: string;
+};
 
 @Injectable()
 export class BetasDaoService {
+  static allGraphs = '[gym,color,wall,gymGrade,creatorProfile]';
+
   constructor(@Inject('BetaModel') private betaModel: ModelClass<BetaModel>) {}
 
-  create(
-    beta: CreateBetaDto & {
-      creatorId: string;
-    },
-  ) {
+  create(beta: CreateBetaQuery) {
     return this.betaModel.query().insert(beta).returning('*');
   }
+
+  private buildGetQuery(args: GetBetasQuery) {
+    let query = this.betaModel.query();
+    if (args.gymId !== undefined) {
+      query = query.where('gymId', args.gymId);
+    }
+    if (args.colorId !== undefined) {
+      query = query.where('colorId', args.colorId);
+    }
+    if (args.gymGradeId !== undefined) {
+      query = query.where('gymGradeId', args.gymGradeId);
+    }
+    if (args.wallId !== undefined) {
+      query = query.where('wallId', args.wallId);
+    }
+    if (args.creatorId !== undefined) {
+      query = query.where('creatorId', args.creatorId);
+    }
+    return query;
+  }
```

Comment:
can simplify to `query.where({...args})`

Reply:
Nice ok!

Reply:
Tested it out. Realised, it's not so trivial b/c GetBetasQuery have other info that needs to be filtered out like page data.

Also running where({ colorId: undefined, ... }) will throw an error.